pkg/controllers/nodeclass/status/ami.go

```diff
@@ -0,0 +1,58 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package status
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"time"
+
+	"github.com/samber/lo"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/aws/karpenter-provider-aws/pkg/apis/v1beta1"
+	"github.com/aws/karpenter-provider-aws/pkg/providers/amifamily"
+)
+
+type AMI struct {
+	amiProvider amifamily.Provider
+}
+
+func (a *AMI) Reconcile(ctx context.Context, nodeClass *v1beta1.EC2NodeClass) (reconcile.Result, error) {
+	amis, err := a.amiProvider.Get(ctx, nodeClass, &amifamily.Options{})
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	if len(amis) == 0 {
+		nodeClass.Status.AMIs = nil
+		return reconcile.Result{}, fmt.Errorf("no amis exist given constraints")
+	}
+	nodeClass.Status.AMIs = lo.Map(amis, func(ami amifamily.AMI, _ int) v1beta1.AMI {
+		reqs := ami.Requirements.NodeSelectorRequirements()
+		sort.Slice(reqs, func(i, j int) bool {
+			if len(reqs[i].Key) != len(reqs[j].Key) {
```

Comment:
Not something that you need to change in this PR, but seems odd to me that we compare the length here first before ordering them alphabetically. Do you know why we had originally decided to do that?

Reply:
We wanted to make sure that architecture came before gpu requirements